File: Test/student_solution/java/Buyer.java
import java.io.*;
class Buyer
{
  private int cash;

  public Buyer(int cash)
  {
    this.cash=cash;
  }

  public int getCash()
  {
    return cash;
  }

  public void setCash(int cash)
  {
    this.cash=cash;
  }

  public void buy(Seller x) throws Exception
  {
    if(cash>0)
    {
      if(x.sell()==1)
      {
        cash--;
      }
    }
	throw new Exception();
  }
}

File: Test/author_solution/cpp/Buyer.cpp
#include "Buyer.hpp"

Buyer::Buyer(int cash)
{
    this->cash=cash;
}

int Buyer::getCash()
{
    return cash;
}

void Buyer::setCash(int cash)
{
    this->cash=cash;
}

void Buyer::buy(Seller x)
{
    if(cash>0)
    {
      if(x.sell()==1)
      {
        cash--;
      }
    }
}

